core/translate.rs

```diff
@@ -272,11 +312,15 @@ fn translate_tables_begin(program: &mut ProgramBuilder, select: &mut Select) {
     }
 }
 
-fn translate_tables_end(program: &mut ProgramBuilder, select: &Select) {
+fn translate_tables_end(program: &mut ProgramBuilder, select: &Select) -> usize {
     // iterate in reverse order as we open cursors in order
+    let mut next_offset = std::usize::MAX;
     for table_loop in select.loops.iter().rev() {
         let cursor_id = table_loop.open_cursor;
         program.emit_insn(Insn::NextAsync { cursor_id });
+        if next_offset == std::usize::MAX {
+            next_offset = program.offset() - 1;
```

Comment:
this was sort of a yolo solution, can we always jump here when WHERE evaluates to false for a row.....?